runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -1185,9 +1185,13 @@ function M.set_loclist(opts)
   local items = {}
   local insert_diag = function(diag)
     local pos = diag.range.start
+    -- Offset for diagnostic reported on terminating EOL character
     local row = pos.line
-    local col = util.character_offset(bufnr, row, pos.character)
+    if row == vim.fn.line('$') then
```

Comment:
Maybe `nvim_buf_line_count` is better here ?

Reply:
Done